x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/Mul.java

```diff
@@ -8,10 +8,54 @@
 package org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic;
 
 import org.elasticsearch.compute.ann.Evaluator;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.arithmetic.BinaryComparisonInversible;
+import org.elasticsearch.xpack.ql.tree.NodeInfo;
+import org.elasticsearch.xpack.ql.tree.Source;
 
+import static org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.EsqlArithmeticOperation.OperationSymbol.MUL;
 import static org.elasticsearch.xpack.ql.util.NumericUtils.unsignedLongMultiplyExact;
 
-public class Mul {
+public class Mul extends EsqlArithmeticOperation implements BinaryComparisonInversible {
```

Comment:
question (out of scope of this PR, though): why is the interface called `BinaryComparisonInversible`? There is only 1 usage of the interface for some optimization rule, but it's unclear to me how it relates to comparisons.